### PR TITLE
GC included txns

### LINKED ISSUE #151: Garbage Collect Included Txns
Currently, we don't have mechanism to GC included_txns, therefore it will grow with time. We ideally want to GC older txns. 

Zulip Discussion: https://espresso.zulipchat.com/#narrow/stream/419378-Block-Builder/topic/Staging.20performance.20issues/near/440153803

### PR DESCRIPTION
**Pull request by move47**
Closes #151

### CODE REVIEW

File: src/builder_state.rs [hunk 1]
```diff
@@ -579,11 +579,22 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
             .for_each(|txn| {
                 if let Entry::Occupied(txn_info) = self.tx_hash_to_available_txns.entry(*txn) {
                     self.timestamp_to_tx.remove(&txn_info.get().0);
-                    self.included_txns.insert(*txn);
+                    self.included_txns.insert(*txn_info.key(), txn_info.get().0);
                     txn_info.remove_entry();
                 }
             });
 
+        // clear the txns from included txns which are older than current timestamp by 3 secs
```
Comment (nyospe) [inline src/builder_state.rs:587]: What? No. I meant, we could clear after... days? Hours, maybe, at the smallest increment.

Comment (nyospe) [reply src/builder_state.rs:587]: We can do 2 hours. But... maybe let's try this instead: https://espresso.zulipchat.com/#narrow/stream/419378-Block-Builder/topic/Staging.20performance.20issues/near/440153803 with the <unit of time> configurable, defaulting to 1 hour.

File: src/builder_state.rs [hunk 2]
```diff
@@ -941,13 +956,33 @@ impl<TYPES: NodeType> BuilderState<TYPES> {
...
+        // txn garbage collection
+        if Instant::now() >= self.next_txn_garbage_collect_time {
+            included_txns_expiring.clone_from(&included_txns_old);
+            included_txns_old.clone_from(&included_txns);
+            included_txns.clear();
+
+            next_txn_garbage_collect_time =
+                self.next_txn_garbage_collect_time + self.txn_garbage_collect_duration;
+        }
+
```
Comment (nyospe) [inline src/builder_state.rs:979]: More idiomatic as:

```rust
let (included_txns, included_txns_old, included_txns_expiring, next_txn_garbage_collect_time) = if Instant::now() >= self.next_txn_garbage_collect_time {
            (HashSet::new(), self.included_txns.clone(), self.included_txns_old.clone(), Instant::now() + self.txn_garbage_collect_duration)
} else {
            (self.included_txns.clone(), self.included_txns_old.clone(), self.included_txns_expiring.clone(), self. next_txn_garbage_collect_time)
};
```

**Review by nyospe — CHANGES_REQUESTED** [review]
One purely stylistic request.

Otherwise, looks good.

File: src/builder_state.rs (hunk 2 above)
Comment (move47) [reply src/builder_state.rs:985]: https://github.com/EspressoSystems/hotshot-builder-core/pull/152/commits/bdacf4f49117f5e29163b768077151022eaf7e25

**Review by nyospe — APPROVED** [review]
LGTM

**Pull request merged**